### PR TITLE
Add `Scan QR Code from Image` action

### LINKED ISSUE #24: Action proposal: Scan QR Code from Image
### Description

iOS Shortcuts does offer `Scan QR Code` but does not accept image input. This `Scan QR Code from Image` will enable us to scan any QR codes from Photo albums and from Share Extension.

Screen Shot 2022-01-20 at 16 38 39

Actually, I've already implemented it and it's (almost) ready for a PR; I'd like to know if this action is good enough to be included in Actions.

### PR DESCRIPTION
**Pull request by uetchy**
This `Scan QR Code from Image` will enable us to scan any QR codes from Photo albums and from Share Extension.

Screen Shot 2022-01-20 at 16 38 39Screen Shot 2022-03-10 at 19 07 06

## ToDo

- [x] Implement the logic
- [x] Tested on iOS Simulator
- [x] Tested on macOS
- [x] Add the action to readme.md and app-store-description.txt
- [x] Make sure linting passes

Fixes #24

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -15,7 +15,7 @@
 The app provides additional actions for the Shortcuts app on macOS and iOS.
 
 [**Submit action idea**](https://github.com/sindresorhus/Actions/issues/new/choose)\
-<sub>*(Submit an issue before [submitting a pull request](.github/contributing.md))*</sub>
+<sub>_(Submit an issue before [submitting a pull request](.github/contributing.md))_</sub>
```
Comment (sindresorhus) [inline readme.md:18]: Don't make unrelated changes.

**Comment by sindresorhus** [comment]
> Make sure linting passes

^

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
The action is missing a description in the Intents editor.

**Comment by sindresorhus** [comment]
The file should be added to the bottom in the file list in Xcode.

**Comment by sindresorhus** [comment]
The action title is not correct titlecase. "From" should be lowercase.

**Comment by sindresorhus** [comment]
Is returning an error when no QR code is found the right choice? Shortcuts does not have any error handling, so it makes it impossible to create shortcuts where the image may or may not have a QR code.

File: IntentsExtension/Actions/ScanQRCodeFromImage.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                                    context: nil,
+                                    options: [CIDetectorAccuracy: CIDetectorAccuracyHigh]),
+          let ciImage = CIImage(data: image) else { return .failure(failure: "Failed to initialize QR Code scanner.") }
+
+    guard let features = detector.features(in: ciImage) as? [CIQRCodeFeature],
+          let message = features.first?.messageString else { return .failure(failure: "No QR Code found in the image.") }
+
+    response.result = message
+
+    return response
```
Comment (sindresorhus) [inline IntentsExtension/Actions/ScanQRCodeFromImage.swift:24]: Use tab-indentation and follow the existing code-style.

File: readme.md (hunk 1 above)
Comment (uetchy) [reply readme.md:18]: Woops I accidentally performed prettier here. Fixed.

File: IntentsExtension/Actions/ScanQRCodeFromImage.swift (hunk 2 above)
Comment (uetchy) [reply IntentsExtension/Actions/ScanQRCodeFromImage.swift:24]: Fixed. Would be nice to have some config file for SwiftFormat or swift-format.

**Comment by uetchy** [comment]
After filling out a description for the action, I got some automatically-generated changes from Xcode. Should I commit these changes?

Screen Shot 2022-03-10 at 19 49 03
Screen Shot 2022-03-10 at 19 49 06

**Comment by uetchy** [comment]
> Is returning an error when no QR code is found the right choice? Shortcuts does not have any error handling, so it makes it impossible to create shortcuts where the image may or may not have a QR code.
You are right. I'll change it to return nil when no QR Code is found.

**Comment by sindresorhus** [comment]
Thanks. I just did a release, so it will be a couple of weeks until this is released.

**Comment by sindresorhus** [comment]
It's out now: https://github.com/sindresorhus/Actions/releases/tag/v1.9.0

**Review by Jamstar00 — COMMENTED** [review]
Good
